ui/dashboards/src/context/DashboardProvider.tsx

```diff
@@ -49,33 +52,56 @@ export function usePanels() {
 }
 
 export function useLayouts() {
-  const { layouts, setLayouts } = useStore(({ layouts, setLayouts }) => ({ layouts, setLayouts }));
-  return { layouts, setLayouts };
+  const { layouts, setLayouts, addLayout, addItemToLayout } = useStore(
+    ({ layouts, setLayouts, addLayout, addItemToLayout }) => ({
+      layouts,
+      setLayouts,
+      addLayout,
+      addItemToLayout,
+    })
+  );
+  return { layouts, setLayouts, addLayout, addItemToLayout };
 }
 
 export function useEditMode() {
   const { isEditMode, setEditMode } = useStore(({ isEditMode, setEditMode }) => ({ isEditMode, setEditMode }));
   return { isEditMode, setEditMode };
 }
 
+export function useDashboard() {
+  const getDashboard = useStore((state) => state.getDashboard);
+  return { getDashboard };
+}
```

Comment:
I might be missing something, but why not return the dashboard itself, vs. a function to call to get the dashboard?

I ask because I noticed that in other cases, like useLayouts and usePanels, we just get the object back, vs. we get a getter and then call the getter.

I'd be down with either pattern of "getting", just wondering if it can be the same pattern everywhere.

Reply:
updated it to return the dashboard itself